### PR TITLE
fix: [CDS-69175]: added regex to support decoded url in http step url validation

### PR DESCRIPTION
**Pull request by satyam-harness**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by satyam-harness** [comment]
retrigger cypress-cd

**Comment by satyam-harness** [comment]
retrigger cypress-rest

**Comment by satyam-harness** [comment]
retrigger cypress-pipeline

**Comment by satyam-harness** [comment]
retrigger coverage

**Comment by satyam-harness** [comment]
retrigger titlecheck

**Review by Sagar-Rajput7 — APPROVED** [review]

File: src/modules/75-cd/components/PipelineSteps/HttpStep/HttpStepWidget.tsx [hunk 1]
```diff
@@ -57,8 +57,8 @@ export function HttpStepWidget(
           url: Yup.lazy((value): Yup.Schema<unknown> => {
             if (getMultiTypeFromValue(value as any) === MultiTypeInputType.FIXED) {
               return Yup.string()
+                .matches(/((https?):\/\/)?(www.)?[a-z0-9]+\..*$/, getString('validation.urlIsNotValid'))
```
Comment (ishantmehtaharness) [inline src/modules/75-cd/components/PipelineSteps/HttpStep/HttpStepWidget.tsx:60]: `((https?):\/\/)?(www.)?[a-z0-9]+\..*$/` - Do we have this regex anywhere in the platform, If yes, we should reuse it

Comment (ishantmehtaharness) [reply src/modules/75-cd/components/PipelineSteps/HttpStep/HttpStepWidget.tsx:60]: If not, we should make it a constant and use it

**Review by ishantmehtaharness — APPROVED** [review]

**Comment by satyam-harness** [comment]
retrigger cypress-pipeline

File: src/modules/75-cd/components/PipelineSteps/HttpStep/HttpStepWidget.tsx (hunk 1 above)
Comment (satyam-harness) [reply src/modules/75-cd/components/PipelineSteps/HttpStep/HttpStepWidget.tsx:60]: this is not used anywhere in the code as of now.

**Pull request merged**